text/0000-pack-inspect-image.md

```diff
@@ -0,0 +1,96 @@
+    NAME                VERSION
+    buildpack.id        0.1.0
+    other.buildpack     2.0.0
+
+Bill of Materials:
+
+  some-dependency:
+    version: "some version"
+  other-dependency
+    version: "other version"
+    metadata:
+      some-key: "some-value"
+      structured-data:
+        foo: "bar"
+        blah: ["blah", "blah"]
+```
+
+By default the `pack` will assume it is inspecting a daemon image. `pack inspect-image <image-ref> --remote` will inspect an image in a registry.
+
+The "Base Image" section refers to the specific run image that was used during build, the "Stack" section includes metadata that may be used during a `pack rebase`
+
+# How it Works
+[how-it-works]: #how-it-works
+
+`exporter` will add the bill-of-materials and the process types to the `io.buildpacks.lifecycle.metadata` label by adding `bom` and `processes` keys respectively.
+
+`pack` will read all necessary information from the metadata label and display it to the user. 
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* Given that the BOM is not guaranteed to be comprehensive, users may be confused if dependencies that are added to the final image are not displayed in the Bill of Materials.
+* Give that the BOM is not currently displayed anywhere. not all buildpacks are writing metadata to the BOM. If most buildpacks do not add entry claims with metadata, the Bill of Materials section of the output may be useless and/or actively misleading
+* The two types of run image information may be confusing.
+* The BOM does not currently associate dependencies with the buildpacks that contributed them.
+* build plan entries are primarily functional, if buildpack authors do not explicitly consider the BOM, the fields they add may not be friendly to human eyes
+
+# Alternatives
+[alternatives]: #alternatives
+
+Currently the image metadata contains all the layer metadata generated during build. This could be displayed instead of the BOM. The advantages to using the layer metadata are
+* all of the buildpacks currently provide layer metadata
+* the layer metadata is already stored in a way that associates it with the contributing buildpack
+* using layer metadata would allow us to tell advanced users specifically which layers dependencies live in if we wished to at a later date
+
+If we chose to use the layer metadata instead, I would propose removing the `[bom]` section from the image `metadata.toml` file, given that it currently serves no compelling purpose.
+
+The problem with either of these approaches is that buildpack authors are likely to view both the build plan entry claims and the layer metadata from a functional rather than an informative perspective. Data in these field may not be human friendly. 
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Should we add the buildpack id and version to the BOM entries?
+- Is there a better format for the BOM output than YAML? TOML would be in keeping with our ecosystem but it's harder to read.
```

Comment:
It's my understanding that BOMs can become quite large. I wonder how we could mitigate this? Perhaps if a BOM is longer than a certain number of lines, we dump it to a file instead and give a link?

Otherwise, I think this new command makes a lot of sense.